### PR TITLE
Issue #4364: Design a new format of suppression dtd schema to support XPath queries

### PR DESCRIPTION
**Pull request by timurt**
https://github.com/checkstyle/checkstyle/issues/4364

### CODE REVIEW

**Comment by rnveach** [comment]
https://github.com/checkstyle/checkstyle/issues/4364#issuecomment-306275951
> 1. Please add the new element to the suppression dtd schema;
> 2. Bump schema version;

I thought we were extending existing schema?
If we make brand new DTD type, than it seems we don't need to rename `suppress` tag need to be `suppress-xpath`? You are basically double defining it as an xpath. By the DTD declaration and the tag name.

Also, should we really merge this DTD by itself even though we aren't using it for anything yet?
Why don't we wait on a basic code implementation to make sure there are no hidden issues before merging? I don't see how delaying can hurt anything.

**Comment by MEZk** [comment]
> I thought we were extending existing schema?

1. The existing suppression dtd schema should be extended. @timurt please update the format of the existing one.

> Also, should we really merge this DTD by itself even though we aren't using it for anything yet?

2. It will not affect any code or SuppressionLoader as the version of the schema will be 1.2. The next step will be https://github.com/checkstyle/checkstyle/issues/4421 and https://github.com/checkstyle/checkstyle/issues/4369. We agreed on the element name and its attributes, that is why I believe that we can add the schema. The format of the new element is pretty similar to ```suppression``` element, that is why I think that there will not be any problems/typos.

**Comment by romani** [comment]
@timurt ,
please read my comment at https://github.com/checkstyle/checkstyle/issues/4364#issuecomment-306787811 for name of dtd.

File: src/main/resources/com/puppycrawl/tools/checkstyle/suppressions_1_1_xpath_experimental.dtd [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+<!ATTLIST suppress files CDATA #REQUIRED
+                   checks CDATA #IMPLIED
+                   id CDATA #IMPLIED
+                   lines CDATA #IMPLIED
+                   columns CDATA #IMPLIED>
+
+<!ELEMENT suppress-xpath EMPTY>
+<!ATTLIST suppress-xpath files CDATA #REQUIRED
+                   checks CDATA #IMPLIED
+                   id CDATA #IMPLIED
```
Comment (MEZk) [inline src/main/resources/com/puppycrawl/tools/checkstyle/suppressions_1_1_xpath_experimental.dtd]: Why do we need id?

Comment (rnveach) [inline src/main/resources/com/puppycrawl/tools/checkstyle/suppressions_1_1_xpath_experimental.dtd]: How else are you going to distinguish suppressing a specific check when multiple instances of the check are defined?

Comment (MEZk) [inline src/main/resources/com/puppycrawl/tools/checkstyle/suppressions_1_1_xpath_experimental.dtd]: Forgot about the fact that we can have multiple modules configured.

@timurt 
Ignore my comment

**Comment by rnveach** [comment]
Feel free to merge @romani.

**Pull request merged**